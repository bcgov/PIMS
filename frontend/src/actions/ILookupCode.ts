export interface ILookupCode {
  code: string;
  name: string;
  id: string;
  isDisabled: boolean;
  isPublic?: boolean;
  isVisible?: boolean;
  type: string;
  parentId?: number;
  sortOrder?: number;
}
