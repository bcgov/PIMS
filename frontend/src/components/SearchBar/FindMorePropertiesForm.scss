@import '../../colors.scss';

#find-more-properties {
  width: 52em;
  .form-section {
    margin: 1em 0;
    padding: 0.5em;
    background-color: $table-header-color;
    border-radius: 5px;
    .form-row,
    .form-row-centred {
      display: flex;
      justify-content: left;
      align-items: center;
      padding: 0 5%;
      .form-label {
        margin: 0.5em;
        width: fit-content;
        text-align: right;
      }
      .form-control.number-input {
        width: 6em;
      }
      .form-control.building-select {
        width: 15em;
      }
      .form-control.project-number-input {
        width: 10em;
      }

      .location-label-col {
        width: 5em;
      }
      .location-col {
        width: 17.8em;
      }
      .lot-size-col {
        .form-label {
          width: 4em;
          text-align: left;
        }
      }
      .lot-size-max-col {
        width: 8.8em;
      }
    }
    .form-row-centred {
      justify-content: center;
    }
  }

  .invalid-feedback-container {
    width: 100%;
    display: flex;
    flex-direction: row-reverse;
    text-align: center;
    .invalid-feedback {
      display: block;
    }
  }
  .search-button {
    width: 100%;
    margin: 0.5em 0;
  }
  .vertical-line {
    border-left: 10px solid rgba(96, 96, 96, 0.2);
    height: 1.5rem;
    border-width: 2px;
  }

  .no-padding {
    padding: 0;
  }
}
