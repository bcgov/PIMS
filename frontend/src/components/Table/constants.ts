export const DEFAULT_PAGE_SIZE = 10;
export const DEFAULT_PAGE_SELECTOR_OPTIONS: number[] = [5, 10, 20, 50, 100];
