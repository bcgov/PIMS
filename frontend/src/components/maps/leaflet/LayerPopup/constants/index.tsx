import * as React from 'react';

export const MUNICIPALITY_LAYER_URL =
  'https://openmaps.gov.bc.ca/geo/pub/WHSE_LEGAL_ADMIN_BOUNDARIES.ABMS_MUNICIPALITIES_SP/wfs?SERVICE=WFS&REQUEST=GetFeature&VERSION=1.3.0&outputFormat=application/json&typeNames=pub:WHSE_LEGAL_ADMIN_BOUNDARIES.ABMS_MUNICIPALITIES_SP';
export const PARCELS_LAYER_URL =
  'https://openmaps.gov.bc.ca/geo/pub/WHSE_CADASTRE.PMBC_PARCEL_FABRIC_POLY_SVW/wfs?service=WFS&REQUEST=GetFeature&VERSION=1.3.0&outputFormat=application/json&typeNames=pub:WHSE_CADASTRE.PMBC_PARCEL_FABRIC_POLY_SVW';

export const parcelLayerPopupConfig = {
  PARCEL_NAME: { label: 'Parcel Name:', display: (data: any) => data.PARCEL_NAME },
  PIN: { label: 'Parcel PIN:', display: (data: any) => data.PIN },
  PID: { label: 'Parcel PID:', display: (data: any) => data.PID },
  PLAN_NUMBER: { label: 'Plan Number:', display: (data: any) => data.PLAN_NUMBER },
  PARCEL_CLASS: { label: 'Parcel Class:', display: (data: any) => data.PARCEL_CLASS },
  MUNICIPALITY: { label: 'Municipality:', display: (data: any) => data.MUNICIPALITY },
  REGIONAL_DISTRICT: {
    label: 'Regional District:',
    display: (data: any) => data.REGIONAL_DISTRICT,
  },
  FEATURE_AREA_SQM: {
    label: 'Area:',
    display: (data: any) => (
      <>
        {data.FEATURE_AREA_SQM} m<sup>2</sup>
      </>
    ),
  },
};

export const municipalityLayerPopupConfig = {
  ADMIN_AREA_GROUP_NAME: {
    label: 'Administration Area:',
    display: (data: any) => `${data.ADMIN_AREA_GROUP_NAME} (${data.CHANGE_REQUESTED_ORG})`,
  },
};
