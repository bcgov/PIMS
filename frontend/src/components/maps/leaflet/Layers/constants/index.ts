export const MUNICIPALITY_LAYER_URL =
  'https://openmaps.gov.bc.ca/geo/pub/WHSE_LEGAL_ADMIN_BOUNDARIES.ABMS_MUNICIPALITIES_SP/wfs?SERVICE=WFS&REQUEST=GetFeature&VERSION=1.3.0&outputFormat=application/json&typeNames=pub:WHSE_LEGAL_ADMIN_BOUNDARIES.ABMS_MUNICIPALITIES_SP';
export const PARCELS_LAYER_URL =
  'https://openmaps.gov.bc.ca/geo/pub/WHSE_CADASTRE.PMBC_PARCEL_FABRIC_POLY_SVW/wfs?service=WFS&REQUEST=GetFeature&VERSION=1.3.0&outputFormat=application/json&typeNames=pub:WHSE_CADASTRE.PMBC_PARCEL_FABRIC_POLY_SVW';

export const parcelLayerPopupConfig = {
  PIN: { label: 'Parcel PIN:', display: (data: any) => data.PIN },
  PID: { label: 'Parcel PID:', display: (data: any) => data.PID },
  REGIONAL_DISTRICT: {
    label: 'Regional District:',
    display: (data: any) => data.REGIONAL_DISTRICT,
  },
};

export const municipalityLayerPopupConfig = {
  ADMIN_AREA_NAME: { label: 'Area Name:', display: (data: any) => data.ADMIN_AREA_NAME },
  ADMIN_AREA_ABBREVIATION: {
    label: 'Area Abbrevation:',
    display: (data: any) => data.ADMIN_AREA_ABBREVIATION,
  },
  ADMIN_AREA_GROUP_NAME: {
    label: 'Area Group Name:',
    display: (data: any) => `${data.ADMIN_AREA_GROUP_NAME} (${data.CHANGE_REQUESTED_ORG})`,
  },
};
