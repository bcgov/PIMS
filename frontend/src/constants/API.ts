// Network URL's
import queryString from 'query-string';
import { AccessRequestStatus } from './accessStatus';

// Generic Params
export interface IPaginateParams {
  page: number;
  quantity?: number;
  sort?: string | string[];
}

export interface IGetUsersParams extends IPaginateParams {
  username?: string;
  firstName?: string;
  lastName?: string;
  email?: string;
  agency?: string;
  role?: string;
  isDisabled?: boolean;
  position?: string;
}

export interface IGetAgenciesParams extends IPaginateParams {
  name?: string;
  description?: string;
  isDisabled?: boolean;
}

export interface IPaginateAccessRequests extends IPaginateParams {
  status?: AccessRequestStatus | null;
}

// Parcels
export interface IPropertySearchParams {
  pid?: string | null;
  neLatitude: number;
  neLongitude: number;
  swLatitude: number;
  swLongitude: number;
  address: string | null;
  administrativeArea: string | null;
  projectNumber: string | null;
  /** comma-separated list of agencies to filter by */
  agencies: string | null;
  classificationId: number | null;
  minLandArea: number | null;
  maxLandArea: number | null;
  inSurplusPropertyProgram: boolean | null;
  inEnhancedReferralProcess?: boolean;
}
export const PROPERTIES = (params: IPropertySearchParams | null) =>
  `/properties/search?${params ? queryString.stringify(params) : ''}`; // get filtered properties or all if not specified.

export const PARCELS_DETAIL = (params: IPropertySearchParams | null) => {
  return `/properties/parcels?${params ? queryString.stringify(params) : ''}`; // get filtered properties or all if not specified.
};
export interface IParcelDetailParams {
  id: number;
}

export const PARCEL_DETAIL = (params: IParcelDetailParams) => `/properties/parcels/${params.id}`;
export const PARCEL_ROOT = `/properties/parcels`;

export interface IUserDetailParams {
  id: string;
}

export interface IAgencyDetailParams {
  id: string;
}

export const AGENCY_ROOT = () => `/admin/agencies/`;
export const AGENCY_DETAIL = (params: IAgencyDetailParams) => `/admin/agencies/${params.id}`;
export const USER_DETAIL = (params: IUserDetailParams) => `/admin/users/${params.id}`;
export const KEYCLOAK_USER_UPDATE = (params: IUserDetailParams) => `/keycloak/users/${params.id}`;

export interface IBuildingDetailParams {
  id: number;
}
export const BUILDING_DETAIL = (params: IBuildingDetailParams) =>
  `/properties/buildings/${params.id}`;

// Lookup Codes
export const LOOKUP_CODE = () => `/lookup`;
export const LOOKUP_CODE_SET = (codeSetName: string) => `/lookup/${codeSetName}`; // get filtered properties or all if not specified.
export const AGENCY_CODE_SET_NAME = 'Agency';
export const ROLE_CODE_SET_NAME = 'Role';
export const PROVINCE_CODE_SET_NAME = 'Province';
export const AMINISTRATIVE_AREA_CODE_SET_NAME = 'AdministrativeArea';
export const PROPERTY_CLASSIFICATION_CODE_SET_NAME = 'PropertyClassification';
export const CONSTRUCTION_CODE_SET_NAME = 'BuildingConstructionType';
export const PREDOMINATE_USE_CODE_SET_NAME = 'BuildingPredominateUse';
export const OCCUPANT_TYPE_CODE_SET_NAME = 'BuildingOccupantType';

// Agencies
export const POST_AGENCIES = () => `/admin/agencies/filter`; // get paged list of agencies

// Auth Service
export const ACTIVATE_USER = () => `/auth/activate`; // get filtered properties or all if not specified.

// User Service
export const REQUEST_ACCESS = (id?: number) => `/users/access/requests${id ? '/' + id : ''}`; //request access url.
export const REQUEST_ACCESS_ADMIN = () => `/keycloak/users/access/request`; //request access admin url.
export const REQUEST_ACCESS_LIST = (params: IPaginateAccessRequests) =>
  `/admin/access/requests?${queryString.stringify(params)}`; // get paged access requests
export const REQUEST_ACCESS_DELETE = (id: number) => `/admin/access/requests/${id}`; // delete an access request
export const POST_USERS = () => `/admin/users/my/agency`; // get paged list of users

// Projects
export const PROJECT_DISPOSE_ROOT = '/projects/disposal/';
export const PROJECT_DISPOSE_WORKFLOW = (code: string) => `/projects/workflows/${code}/status`;
export const PROJECT_DISPOSE_TASKS = (code: string | number) => `/projects/status/${code}/tasks`;
export const PROJECT_STATUSES = '/projects/status';
export const PROJECT_WORKFLOW_TASKS = (code: string) => `/projects/workflows/${code}/tasks`;
export const PROJECT_UPDATE_WORKFLOW_STATUS = (workflowCode: string, statusCode: string) =>
  `${PROJECT_DISPOSE_ROOT}workflows/${workflowCode}/${statusCode}`;
