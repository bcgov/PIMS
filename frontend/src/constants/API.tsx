// Network URL's
import queryString from "query-string";

// Parcels
export interface IParcelListParams {
  neLat: number,
  neLong: number,
  swLat: number,
  swLong: number,
  agencyId: number | null,
  propertyClassificationId: number | null
}
export const PARCELS = (params: IParcelListParams | null) => (params ? `/my/parcel?${queryString.stringify(params)}` : "/my/parcel"); // get filtered properties or all if not specified.
export interface IParcelDetailParams {
  id: number
}
export const PARCELDETAIL = (params: IParcelDetailParams) => (`/my/parcel/${params.id}`);

// Lookup Codes
export const LOOKUP_CODE_SET = (codeSetName: string) => (`/lookup/${codeSetName}`); // get filtered properties or all if not specified.
export const AGENCY_CODE_SET_NAME = "agency";
export const PROPERTY_CLASSIFICATION_CODE_SET_NAME = "propertyClassification";