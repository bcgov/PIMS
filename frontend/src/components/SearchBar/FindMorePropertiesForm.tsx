import './FindMorePropertiesForm.scss';

import { Button, Check, DisplayError, Input, Select } from 'components/common/form';
import { TypeaheadField } from 'components/common/form/Typeahead';
import * as API from 'constants/API';
import { getIn, useFormikContext } from 'formik';
import useCodeLookups from 'hooks/useLookupCodes';
import React, { ReactElement, useEffect, useState } from 'react';
import { Col, Container, Form, Row } from 'react-bootstrap';
import { mapLookupCode } from 'utils';

/** bar separator for the inputs */
const VerticalLine = () => <span className="vertical-line" />;

/** the area where invalid feedback or ERP/SPL selection will be displayed */
interface IChildProps {
  children: ReactElement;
}
const InvalidFeedback = ({ children }: IChildProps) => (
  <div className="invalid-feedback-container">{children}</div>
);

/** This form is triggered by the FindMorePropertiesButton and contains additional filter fields for the PropertiesFilter */
const FindMorePropertiesForm = () => {
  const lookupCodes = useCodeLookups();
  const { setFieldValue, handleSubmit, errors } = useFormikContext<any>();
  const [clear, setClear] = useState(false);
  const [displayError, setDisplayError] = useState(false);

  const predominateUses = lookupCodes
    .getByType(API.PREDOMINATE_USE_CODE_SET_NAME)
    .map(mapLookupCode);
  const adminAreas = lookupCodes
    .getByType(API.AMINISTRATIVE_AREA_CODE_SET_NAME)
    .map((c) => mapLookupCode(c, null));

  /** attempt submission of search, display errors if present */
  const handleSearch = () => {
    handleSubmit();
    if (errors) {
      setDisplayError(true);
    }
  };

  useEffect(() => {
    // Track <a/> tag clicks in Snowplow Analytics.
    window.snowplow('refreshLinkClickTracking');
  }, []);

  return (
    <div id="find-more-properties">
      <p>
        Search for properties within the Enhanced Referral Process (ERP) or Surplus Property List
        (SPL)
      </p>
      <p>
        <strong>Note: </strong> The search results will
        <strong>
          &nbsp;only include properties in projects that have been identified as Tier 2, 3, or 4
        </strong>
        <br></br> in the disposal process. For properties included in Tier 1, please contact
        <em> Strategic Real Estate Services</em> at<br></br>
        <a href="mailto:RealPropertyDivision.Disposals@gov.bc.ca">
          RealPropertyDivision.Disposals@gov.bc.ca
        </a>
      </p>
      <Container className="form-section">
        {displayError && (
          <InvalidFeedback>
            <DisplayError errorPrompt field="inSurplusPropertyProgram" />
          </InvalidFeedback>
        )}

        <Row className="form-row form-row-centred">
          <Col md="auto">
            <Check label="ERP Properties" field="inEnhancedReferralProcess" />
          </Col>
          <Col md="auto">
            <VerticalLine />
          </Col>
          <Col md="auto">
            <Check label="SPL Properties" field="inSurplusPropertyProgram" />
          </Col>
        </Row>
      </Container>
      <Container className="form-section">
        <Row>
          <h6>Search By</h6>
        </Row>
        <Row className="form-row">
          <Col md="auto" className="location-label-col">
            <Form.Label>Location</Form.Label>
          </Col>
          <Col md="auto" className="location-col">
            <TypeaheadField
              name="administrativeArea"
              placeholder="Enter a location"
              paginate={false}
              hideValidation={true}
              options={adminAreas.map((x: any) => x.label)}
              onChange={(vals: any) => {
                setFieldValue('administrativeArea', getIn(vals[0], 'name') ?? vals[0]);
              }}
              clearSelected={clear}
              setClear={setClear}
            />
          </Col>
          <Col md="auto">
            <VerticalLine />
          </Col>
          <Col md="auto">
            <Input
              field="projectNumber"
              label="Project Number"
              placeholder="SPP #"
              className="project-number-input"
            />
          </Col>
        </Row>
      </Container>
      <Container className="form-section">
        <Row>
          <h6>Land Search Criteria</h6>
        </Row>
        <Row className="form-row">
          <Col md="auto" className="lot-size-col">
            <Input label="Lot Size" field="minLotSize" placeholder="min" className="number-input" />
          </Col>
          <Col md="auto" className="no-padding">
            <span>-</span>
          </Col>
          <Col md="auto" className="lot-size-max-col">
            <Input field="maxLotSize" placeholder="max" className="number-input" />
          </Col>
          <Col md="auto">
            <VerticalLine />
          </Col>
          <Col md="auto">
            <Check label="Land Only" field="bareLandOnly" />
          </Col>
        </Row>
      </Container>
      <Container className="form-section">
        <Row>
          <h6>Building Search Criteria</h6>
        </Row>
        <Row className="form-row">
          <Col md="auto">
            <Select
              className="building-select"
              field="predominateUseId"
              placeholder="Any"
              options={predominateUses}
              label="Predominate Use"
            />
          </Col>
          <Col md="auto">
            <VerticalLine />
          </Col>
          <Col md="auto">
            <Input label="Net Usable Area" field="rentableArea" className="number-input" />
          </Col>
        </Row>
      </Container>
      <Row>
        <Col xs={12}>
          <Button
            className="search-button"
            onClick={() => {
              handleSearch();
            }}
          >
            Search
          </Button>
        </Col>
      </Row>
    </div>
  );
};

export default FindMorePropertiesForm;
