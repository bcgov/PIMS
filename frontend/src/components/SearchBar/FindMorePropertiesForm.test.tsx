import Adapter from '@cfaester/enzyme-adapter-react-18';
import { fireEvent, render, waitFor } from '@testing-library/react';
import { ILookupCode } from 'actions/ILookupCode';
import FindMorePropertiesForm from 'components/SearchBar/FindMorePropertiesForm';
import * as API from 'constants/API';
import Claims from 'constants/claims';
import Enzyme from 'enzyme';
import { getIn, useFormikContext } from 'formik';
import { createMemoryHistory } from 'history';
import useKeycloakWrapper from 'hooks/useKeycloakWrapper';
import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import useKeycloakMock from 'useKeycloakWrapperMock';

const userRoles: string[] | Claims[] = [];
const userAgencies: number[] = [1];
const userAgency: number = 1;

jest.mock('hooks/useKeycloakWrapper');
(useKeycloakWrapper as jest.Mock).mockReturnValue(
  new (useKeycloakMock as any)(userRoles, userAgencies, userAgency),
);

jest.mock('formik');
Enzyme.configure({ adapter: new Adapter() });

const handleSubmit = jest.fn();

(useFormikContext as jest.Mock).mockReturnValue({
  values: {
    classificationId: 0,
    classification: 'zero',
  },
  registerField: jest.fn(),
  unregisterField: jest.fn(),
  handleSubmit: handleSubmit,
});
(getIn as jest.Mock).mockReturnValue(0);

const mockStore = configureMockStore([thunk]);
const history = createMemoryHistory();

const lCodes = {
  lookupCodes: [
    { name: 'agencyVal', id: '1', isDisabled: false, type: API.AGENCY_CODE_SET_NAME },
    { name: 'disabledAgency', id: '2', isDisabled: true, type: API.AGENCY_CODE_SET_NAME },
  ] as ILookupCode[],
};

const selectedUser = {
  username: 'tester',
  firstName: 'firstName',
  lastName: 'lastName',
  email: 'test@test.com',
  isDisabled: false,
  emailVerified: false,
  agencies: [1],
  roles: [],
  rowVersion: 'AAAAAAAAB9E=',
  note: 'test note',
  lastLogin: '2023-02-28T17:45:39.7381599',
};

const store = mockStore({
  users: { user: selectedUser },
  lookupCode: lCodes,
});

const RenderHelper = () => (
  <Provider store={store}>
    <MemoryRouter initialEntries={[history.location]}>
      <FindMorePropertiesForm />
    </MemoryRouter>
  </Provider>
);

describe('Testing FindMorePropertiesForm', () => {
  it('Snapshot Test', () => {
    const tree = renderer.create(<RenderHelper />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('handleSubmit function called', () => {
    const { container } = render(<RenderHelper />);
    const searchButton = container.querySelector('.search-button');
    waitFor(() => {
      fireEvent.click(searchButton!);
    });

    expect(handleSubmit).toHaveBeenCalledTimes(1);
  });
});
