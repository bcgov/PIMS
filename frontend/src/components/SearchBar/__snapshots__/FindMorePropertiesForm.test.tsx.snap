// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing FindMorePropertiesForm Snapshot Test 1`] = `
<div
  id="find-more-properties"
>
  <p>
    Search for properties within the Enhanced Referral Process (ERP) or Surplus Property List (SPL)
  </p>
  <p>
    <strong>
      Note: 
    </strong>
     The search results will
    <strong>
      Â only include properties in projects that have been identified as Tier 2, 3, or 4
    </strong>
    <br />
     in the disposal process. For properties included in Tier 1, please contact
    <em>
       Strategic Real Estate Services
    </em>
     at
    <br />
    <a
      href="mailto:RealPropertyDivision.Disposals@gov.bc.ca"
    >
      RealPropertyDivision.Disposals@gov.bc.ca
    </a>
  </p>
  <div
    className="form-section container"
  >
    <div
      className="form-row form-row-centred row"
    >
      <div
        className="col-md-auto"
      >
        <div
          className=""
        >
          <div
            className="check-field"
          >
            <label
              className="form-label"
              htmlFor="input-inEnhancedReferralProcess"
              style={
                Object {
                  "marginRight": "10px",
                }
              }
            >
              ERP Properties
            </label>
            <div
              className=""
            >
              <input
                checked={false}
                className="form-check-input"
                disabled={false}
                id="input-inEnhancedReferralProcess"
                name="inEnhancedReferralProcess"
                onChange={[Function]}
                type="checkbox"
                value="false"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="col-md-auto"
      >
        <span
          className="vertical-line"
        />
      </div>
      <div
        className="col-md-auto"
      >
        <div
          className=""
        >
          <div
            className="check-field"
          >
            <label
              className="form-label"
              htmlFor="input-inSurplusPropertyProgram"
              style={
                Object {
                  "marginRight": "10px",
                }
              }
            >
              SPL Properties
            </label>
            <div
              className=""
            >
              <input
                checked={false}
                className="form-check-input"
                disabled={false}
                id="input-inSurplusPropertyProgram"
                name="inSurplusPropertyProgram"
                onChange={[Function]}
                type="checkbox"
                value="false"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="form-section container"
  >
    <div
      className="row"
    >
      <h6>
        Search By
      </h6>
    </div>
    <div
      className="form-row row"
    >
      <div
        className="location-label-col col-md-auto"
      >
        <label
          className="form-label"
        >
          Location
        </label>
      </div>
      <div
        className="location-col col-md-auto"
      >
        <div
          className=""
        >
          <div
            onBlur={[Function]}
            onFocus={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
          >
            <div
              className="rbt"
              style={
                Object {
                  "outline": "none",
                  "position": "relative",
                }
              }
              tabIndex={-1}
            >
              <div
                style={
                  Object {
                    "display": "flex",
                    "flex": 1,
                    "height": "100%",
                    "position": "relative",
                  }
                }
              >
                <input
                  aria-autocomplete="both"
                  aria-expanded={false}
                  aria-haspopup="listbox"
                  autoComplete="off"
                  className="rbt-input-main form-control rbt-input"
                  id="administrativeArea-field"
                  name="administrativeArea"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  placeholder="Enter a location"
                  role="combobox"
                  style={
                    Object {
                      "width": "100%",
                    }
                  }
                  type="text"
                  value=""
                />
                <input
                  aria-hidden={true}
                  className="rbt-input-hint"
                  readOnly={true}
                  style={
                    Object {
                      "backgroundColor": "transparent",
                      "borderColor": "transparent",
                      "boxShadow": "none",
                      "color": "rgba(0, 0, 0, 0.54)",
                      "left": 0,
                      "pointerEvents": "none",
                      "position": "absolute",
                      "top": 0,
                      "width": "100%",
                    }
                  }
                  tabIndex={-1}
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="col-md-auto"
      >
        <span
          className="vertical-line"
        />
      </div>
      <div
        className="col-md-auto"
      >
        <div
          className="input-flex"
          id="input-projectNumber"
        >
          <label
            className="form-label"
          >
            Project Number
          </label>
          <div
            onBlur={[Function]}
            onFocus={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
          >
            <input
              className="project-number-input form-control"
              name="projectNumber"
              onBlur={[Function]}
              placeholder="SPP #"
              value={0}
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="form-section container"
  >
    <div
      className="row"
    >
      <h6>
        Land Search Criteria
      </h6>
    </div>
    <div
      className="form-row row"
    >
      <div
        className="lot-size-col col-md-auto"
      >
        <div
          className="input-flex"
          id="input-minLotSize"
        >
          <label
            className="form-label"
          >
            Lot Size
          </label>
          <div
            onBlur={[Function]}
            onFocus={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
          >
            <input
              className="number-input form-control"
              name="minLotSize"
              onBlur={[Function]}
              placeholder="min"
              value={0}
            />
          </div>
        </div>
      </div>
      <div
        className="no-padding col-md-auto"
      >
        <span>
          -
        </span>
      </div>
      <div
        className="lot-size-max-col col-md-auto"
      >
        <div
          className="input-flex"
          id="input-maxLotSize"
        >
          <div
            onBlur={[Function]}
            onFocus={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
          >
            <input
              className="number-input form-control"
              name="maxLotSize"
              onBlur={[Function]}
              placeholder="max"
              value={0}
            />
          </div>
        </div>
      </div>
      <div
        className="col-md-auto"
      >
        <span
          className="vertical-line"
        />
      </div>
      <div
        className="col-md-auto"
      >
        <div
          className=""
        >
          <div
            className="check-field"
          >
            <label
              className="form-label"
              htmlFor="input-bareLandOnly"
              style={
                Object {
                  "marginRight": "10px",
                }
              }
            >
              Land Only
            </label>
            <div
              className=""
            >
              <input
                checked={false}
                className="form-check-input"
                disabled={false}
                id="input-bareLandOnly"
                name="bareLandOnly"
                onChange={[Function]}
                type="checkbox"
                value="false"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="form-section container"
  >
    <div
      className="row"
    >
      <h6>
        Building Search Criteria
      </h6>
    </div>
    <div
      className="form-row row"
    >
      <div
        className="col-md-auto"
      >
        <div
          className="select-row row"
          controlid="input-predominateUseId"
        >
          <div
            className="col-md-auto"
          >
            <label
              className="form-label"
            >
              Predominate Use
            </label>
          </div>
          <div
            className="select-column col-md-auto"
          >
            <select
              className="building-select form-select form-control"
              name="predominateUseId"
              onBlur={[Function]}
              onChange={[Function]}
              value={0}
            >
              <option
                value=""
              >
                Any
              </option>
            </select>
          </div>
        </div>
      </div>
      <div
        className="col-md-auto"
      >
        <span
          className="vertical-line"
        />
      </div>
      <div
        className="col-md-auto"
      >
        <div
          className="input-flex"
          id="input-rentableArea"
        >
          <label
            className="form-label"
          >
            Net Usable Area
          </label>
          <div
            onBlur={[Function]}
            onFocus={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
          >
            <input
              className="number-input form-control"
              name="rentableArea"
              onBlur={[Function]}
              value={0}
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-12"
    >
      <button
        className="Button search-button btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <div
          className="Button__value"
        >
          Search
        </div>
      </button>
    </div>
  </div>
</div>
`;
